src/librustc_trans/back/link.rs

```diff
@@ -1338,7 +1344,13 @@ fn add_upstream_native_libraries(cmd: &mut Linker, sess: &Session, crate_type: c
                 continue
             }
             match lib.kind {
-                NativeLibraryKind::NativeUnknown => cmd.link_dylib(&lib.name.as_str()),
+                NativeLibraryKind::NativeUnknown => {
+                    if sess.crt_static() && !sess.target.target.options.crt_static_allows_dylibs {
+                        cmd.link_staticlib(&lib.name.as_str())
```

Comment:
There's a doc comment on this function that says "Remember that all of these upstream native dependencies are all non-static dependencies." - this isn't true any more.